src/tgbot_expenses/services/account_service.py

```diff
@@ -0,0 +1,124 @@
+    :type account_name: str
+    :param account_amount: The amount associated with the account
+                           to insert.
+    :type account_amount: Decimal
+    :return: None
+    """
+    async with AsyncSessionWithEnter(database.engine) as session:
+        account = Account(name=account_name, balance=account_amount)
+        session.add(account)
+        await session.commit()
+
+    return
+
+
+async def get_amount(account_id: int) -> Decimal:
+    """
+    Retrieve the amount associated with the given account ID from
+    the 'accounts' table.
+
+    :param account_id: The ID of the account to retrieve the amount for.
+    :type account_id: int
+    :return: The balance associated with the account.
+    """
+    async with AsyncSessionWithEnter(database.engine) as session:
+        account_obj = await session.execute(select(Account).where(
+            Account.id == account_id
+        ))
+        account = account_obj.scalars().first()
+        balance = account.balance
+
+        return balance
+
+
+async def get_all_accounts() -> str:
+    """
+    Retrieve the names of all active accounts from the 'accounts' table.
+
+    :return: A string containing the names of all active accounts,
+             separated by semicolons.
+    """
+    async with AsyncSessionWithEnter(database.engine) as session:
+        accounts = await session.execute(select(Account.name).where(
+            Account.account_status == "active"
+        ))
+
+        return ";".join([account[0] for account in accounts])
+
+
+async def update_amount(account_from: str,
+                        amount_old_currency: Decimal,
+                        currency_amount: Decimal, account_to: str) -> None:
+    """
+    Update the amount for the specified account in the 'accounts' table.
+
+    :param account_from: The name of the account to subtract the old
+                         amount from.
+    :type account_from: str
+    :param amount_old_currency: The old amount in the original currency to
+                                subtract from the 'account_from'.
+    :type amount_old_currency: Decimal
+    :param currency_amount: The amount in the new currency to add
+                            to the 'account_to'.
+    :type currency_amount: Decimal
+    :param account_to: The name of the account to add the new amount to.
+    :type account_to: str
+    :return: None
+    """
+    async with AsyncSessionWithEnter(database.engine) as session:
+        async with session.begin():
+            account_obj_from = await session.execute(
+                select(Account).where(
+                    Account.name == account_from
+                ).with_for_update()
+            )
+            account_obj_to = await session.execute(
+                select(Account).where(
+                    Account.name == account_to
+                ).with_for_update()
+            )
+
+            account_from = account_obj_from.scalars().first()
```

Comment:
Will your bot crash if there is no such an account? If provided id is incorrect?

Reply:
Added the 'try`-`except` block